src/openforms/contrib/zgw/clients/catalogi.py

```diff
@@ -227,6 +234,37 @@ def find_case_types(
 
         return all_versions
 
+    def find_case_types_products(
+        self,
+        *,
+        catalogus: str,
+        identification: str,
+        valid_on: date | None = None,
+    ) -> list[Product]:
+        products: list[Product] = []
+        case_versions = self.find_case_types(
+            catalogus=catalogus,
+            identification=identification,
+            valid_on=valid_on,
+        )
+
+        if (
+            case_versions
+            and len(case_versions) > 0
+            and "productenOfDiensten" in case_versions[0]
+        ):
+            for product_uri in case_versions[0]["productenOfDiensten"]:
+                description = ""
+
+                response = requests.get(
+                    product_uri, headers={"Content-Type": "application/json"}
+                )
```

Comment:
Couple remarks here:

1. This is basically server-side request forgery, since the code blindly trusts the URL returned from the zaaktype response
2. using bare `requests` will never add any authorizations configured in the service admin, and will likely return 403 for those product URLs since these APIs are not public
3. we don't send a request body, so `Content-Type` header is not relevant. I think you mean to say `Accept: application/json`

You can resolve 1 and 2 by making use of our client/service libraries: 

```
from zgw_consumers.models import Service
from zgw_consumers.client import build_client

service = Service.get_service(url)
client = build_client(service)
```

`get_service` looks up the URL in the configured services in the DB, so it has all the necessary credentials and configuration. `build_client` understands that information and gives you an [ape-pie](https://pypi.org/project/ape-pie/) client, which is really just a `requests.Session` subclass with the footguns removed.

This means that any API roots that aren't configured in the admin won't yield a service, and we are not vulnerable to SSRF (since we crash/abort instead).

Reply:
Yeah this is 1000% better. I wasn't sure how i could/should do requests, but yeah `requests.get()` was pretty un-safe...

I've changed it to this service/client setup, which seems to work fine